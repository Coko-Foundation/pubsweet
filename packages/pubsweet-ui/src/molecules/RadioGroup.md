A group of radio buttons.

```js
const options = [
  {
    value: 'one',
    label: 'One'
  },
  {
    value: 'two',
    label: 'Two'
  },
  {
    value: 'three',
    label: 'Three'
  }
];

<RadioGroup 
  options={options} 
  name="radiogroup"
  onChange={value => console.log(value)}/>
```

The buttons can be displayed inline

```js
const options = [
  {
    value: 'one',
    label: 'One'
  },
  {
    value: 'two',
    label: 'Two'
  },
  {
    value: 'three',
    label: 'Three'
  }
];

<RadioGroup 
  options={options} 
  name="radiogroup-inline"
  inline={true}
  onChange={value => console.log(value)}/>
```
