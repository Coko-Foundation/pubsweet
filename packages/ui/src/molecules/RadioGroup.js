import React from 'react'

import { Radio, Flexbox } from '../atoms'

class RadioGroup extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      value: props.value,
    }
  }

  handleChange = event => {
    const { value } = event.target
    this.setState({ value })
    this.props.onChange(value)
  }

  render() {
    const { inline, name, options, required } = this.props
    const { value } = this.state

    return (
      <Flexbox inline={!inline}>
        {options.map(option => (
          <Radio
            checked={option.value === value}
            color={option.color}
            inline={inline}
            key={option.value}
            label={option.label}
            name={name}
            onChange={this.handleChange}
            required={required}
            value={option.value}
          />
        ))}
      </Flexbox>
    )
  }
}

export default RadioGroup
