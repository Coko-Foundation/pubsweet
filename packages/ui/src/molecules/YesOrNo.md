A group of radio buttons that provides just two options: "Yes" or "No"

```js
<YesOrNo name="yesorno" onChange={value => console.log(value)} />
```

If a value is set, one option is selected.

```js
<YesOrNo
  name="yesorno-value"
  value="yes"
  onChange={value => console.log(value)}
/>
```

Different statuses have different colours:

```js
<div>
  <YesOrNo
    primary
    name="yesornocoloured"
    onChange={value => console.log(value)}
  />

  <YesOrNo
    warning
    name="yesornocoloured"
    onChange={value => console.log(value)}
  />
</div>
```
