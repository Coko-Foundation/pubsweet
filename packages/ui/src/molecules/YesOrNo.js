import React from 'react'
import RadioGroup from './RadioGroup'

const options = [
  {
    label: 'Yes',
    value: 'yes',
  },
  {
    label: 'No',
    value: 'no',
  },
]

const YesOrNo = ({ name, value, required, onChange, color = 'black' }) => (
  <RadioGroup
    inline
    name={name}
    onChange={onChange}
    options={options.map(el => ({ ...el, color }))}
    required={required}
    value={value}
  />
)

export default YesOrNo
