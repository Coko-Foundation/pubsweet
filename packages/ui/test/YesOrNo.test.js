import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import YesOrNo from '../src/molecules/YesOrNo'
import RadioGroup from '../src/molecules/RadioGroup'
import TestThemeProvider from './setup/theme'

const props = {
  name: 'TestName',
  value: 'Maybe',
}

const wrapper = shallow(<YesOrNo {...props} />)
const radio = wrapper.find(RadioGroup)

describe('Yes or No', () => {
  test('is rendered correclty', () => {
    const tree = renderer
      .create(
        <TestThemeProvider>
          <YesOrNo {...props} />
        </TestThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders a RadioGroup', () => {
    expect(wrapper.is('RadioGroup')).toBeTruthy()
    expect(radio).toHaveLength(1)
  })

  test('Passes the correct options', () => {
    const { options } = radio.props()
    expect(options).toHaveLength(2)

    expect(options[0].value).toEqual('yes')
    expect(options[0].label).toEqual('Yes')

    expect(options[1].value).toEqual('no')
    expect(options[1].label).toEqual('No')
  })

  test('Passes down the correct name', () => {
    const { name } = radio.props()
    expect(name).toEqual(props.name)
  })

  test('Passes down the correct value', () => {
    const { value } = radio.props()
    expect(value).toEqual(props.value)
  })
})
