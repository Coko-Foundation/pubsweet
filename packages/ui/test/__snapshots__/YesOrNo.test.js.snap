// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Yes or No is rendered corrctly 1`] = `
.c2 {
  display: none;
}

.c4 {
  display: inline-block;
  content: ' ';
  width: calc(8px * 2);
  height: calc(8px * 2);
  vertical-align: center;
  margin-left: 8px;
  margin-right: 8px;
  border: calc(8px / 4) solid white;
  border-radius: 50%;
  -webkit-transition: border 0.1s ease;
  transition: border 0.1s ease;
  color: black;
}

.c6 {
  display: inline-block;
  font-family: inherit;
  font-size: 18px;
  font-style: italic;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 1s;
  transition: all 1s;
  min-height: 32px;
  color: black;
}

.c1:not(:last-child) {
  margin-right: 32px;
  margin-bottom: 0;
}

.c1 .c3 {
  background: transparent;
  box-shadow: 0 0 0 1px currentcolor;
}

.c1:hover .c5 {
  color: black;
}

.c1:hover .c3 {
  -webkit-animation-name: kLnLOV;
  animation-name: kLnLOV;
  -webkit-animation-duration: 0.2s;
  animation-duration: 0.2s;
  box-shadow: 0 0 0 1px black;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
}

<div
  className="c0"
>
  <label
    checked={false}
    className="c1"
    color="black"
  >
    <input
      checked={false}
      className="c2"
      disabled={undefined}
      name="TestName"
      onChange={[Function]}
      required={undefined}
      type="radio"
      value="yes"
    />
     
    <span
      className="c3 c4"
      color="black"
    />
     
    <span
      className="c5 c6"
    >
       
      Yes
       
    </span>
     
  </label>
  <label
    checked={false}
    className="c1"
    color="black"
  >
    <input
      checked={false}
      className="c2"
      disabled={undefined}
      name="TestName"
      onChange={[Function]}
      required={undefined}
      type="radio"
      value="no"
    />
     
    <span
      className="c3 c4"
      color="black"
    />
     
    <span
      className="c5 c6"
    >
       
      No
       
    </span>
     
  </label>
</div>
`;
